File: library/src/main/java/me/shkschneider/skeleton/java/RandomHelper.java
package me.shkschneider.skeleton.java;

import android.support.annotation.IntRange;
import android.support.annotation.NonNull;

import java.util.Random;

import me.shkschneider.skeleton.helper.LogHelper;

public class RandomHelper {

    protected RandomHelper() {
        // Empty
    }

    public static boolean binary() {
        return (Math.random() < 0.5);
    }

    public static String string(@NonNull final String characters, @IntRange(from=0) final int length) {
        final Random random = new Random();
        final StringBuilder stringBuilder = new StringBuilder();
        for (int i = 0; i < length; i++) {
            stringBuilder.append(characters.charAt(random.nextInt(characters.length())));
        }
        return stringBuilder.toString();
    }

    public static String string(@IntRange(from=0) final int length) {
        return string(StringHelper.lower(StringHelper.HEX), length);
    }

    public static int inclusive(final int min, final int max) {
        if (min > max) {
            LogHelper.warning("MIN was greater than MAX");
            return -1;
        }
        if (min == max) {
            LogHelper.info("MIN was equal to MAX");
            return min;
        }
        return min + (int) (Math.random() * ((max - min) + 1));
    }

    public static int exclusive(final int min, final int max) {
        return inclusive(min + 1, max - 1);
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/java/RandomHelper.kt
package me.shkschneider.skeleton.java

import android.support.annotation.IntRange
import me.shkschneider.skeleton.helper.LogHelper
import java.util.*

object RandomHelper {

    fun binary(): Boolean {
        return Math.random() < 0.5
    }

    fun string(@IntRange(from = 0) length: Int, characters: String = StringHelper.HEX): String {
        val random = Random()
        val stringBuilder = StringBuilder()
        for (i in 0 until length) {
            stringBuilder.append(characters[random.nextInt(characters.length)])
        }
        return stringBuilder.toString()
    }

    fun inclusive(min: Int, max: Int): Int {
        if (min > max) {
            LogHelper.warning("MIN was greater than MAX")
            return -1
        }
        if (min == max) {
            LogHelper.info("MIN was equal to MAX")
            return min
        }
        return min + (Math.random() * (max - min + 1)).toInt()
    }

    fun exclusive(min: Int, max: Int): Int {
        return inclusive(min + 1, max - 1)
    }

}
